src/readABI.js

```diff
@@ -0,0 +1,35 @@
+import fs from 'fs';
+/**
+ * First tries to read ABI from stdin, then from --abi (-a)
+ * command line parameter.
+ * @param {string} filename user specified or default ABI filename
+ * @returns {Promise<[]>} ABI interface
+ */
+function readABI(filename) {
+  const { stdin } = process;
+  stdin.setEncoding('utf8');
+  return new Promise((resolve, reject) => {
+    // trying to read a file from stdin (command line)
+    stdin.on('data', (file) => {
+      try {
+        resolve(JSON.parse(file));
+      } catch (error) {
+        reject(error);
+      }
+    });
```

Comment:
this assumes that the full file will only be 1 chunk - which is probably OK most of the time - but could fail on larger files, and is dependent on OS configurations.